### ISSUE TITLE
Writing tests using 6to5 code

### ISSUE THREAD

**Issue by jmeas** [6to5/6to5#482]
Hey 6to5 team! Great work. I recently began switching over a big application to 6to5, and I'm curious how to approach switching over the tests.

The old app was an AMD app. The tests loaded the individual AMD modules to test individual pieces. For instance, if I was writing tests for `ArbitraryView`, I would:

```js
var ArbitraryView = require('path/to/arbitrary-view');
```

within the test. Ideally, I'd like to keep this general approach for two reasons. The first is preference. I don't want to bundle the whole app together to test the individual pieces. The other is that there's no way to get access to `ArbitraryView` when the app is bundled together. All of the objects in the app are hidden away and aren't exposed by the built file.

Another thing to consider is that this app's tests can be executed using a headless browser+node.js in addition to using a browser spec runner.

I understand that this may not be the right forum to ask this question. If so, sorry! I feel that this question broadly applicable to the Javascript community, though, because I think my testing setup is likely to be found in many other apps. @kkemple, I know, has similar questions.

So...how do y'all approach testing your es6 apps?

Thanks!

**Comment by benjamingr** [comment]
If you use the require hook (tutorial on the 6to5.org website) you can require an ES6 module into your existing 6to5 code. So if your ArbitraryView is a ES6 exported class now you could do


    var ArbitraryView = require("...").ArbitraryView

Generally that'd justwork.

**Comment by fkling** [comment]
I simply compile all files to ES5 (e.g. from `src/` to `node/`) and run the compiled test in `node/`. My test env setup scripts loads the polyfill (`require('6to5/polyfill')`).

**Comment by monsanto** [comment]
I really don't like worrying about the various entry points to my application (node? browserify? mocha? nodemon? general fiddling around in the repl? etc) so I compile all my files in a build step and prepend `"use strict"; require('my/polyfill')` to each of the compiled files.

This strategy is especially nice if you decide to switch out `6to5/polyfill` for something else. For example, I prefer to use Bluebird instead of the core.js Promise shim.

**Comment by sebmck** [comment]
Closing this as it appears to have been answered. Feel free to reopen or create another issue if it's still an issue. Thanks!

**Comment by jmeas** [comment]
Thanks for the help, everyone! I decided to use something most similar to @fkling's system for testing smaller libraries. I came up with two systems for running tests, one works in Node, the other in the browser. Both of them start with compiling ES6 modules to CommonJS.

Then, for running tests in node, I can simply require the files in. And for running the tests in the browser, I use browserify to bundle up those CommonJS files into a build that's browser-ready.

Works great! Thanks again :tada:
